arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/signals/AbortingConditions.java

```diff
@@ -30,72 +33,79 @@
 public final class AbortingConditions {
   private boolean callStackOverflow;
   private boolean insufficientBalance;
+  private boolean maxNonceReached;
 
   /**
    * @param callStackOverflow too many nested contexts
    * @param insufficientBalance trying to give more ETH than the caller has
+   * @param maxNonceReached the nonce of the creator account is too high (EIP-2681)
    */
-  public AbortingConditions(boolean callStackOverflow, boolean insufficientBalance) {
+  public AbortingConditions(
+      boolean callStackOverflow, boolean insufficientBalance, boolean maxNonceReached) {
     this.callStackOverflow = callStackOverflow;
     this.insufficientBalance = insufficientBalance;
-  }
-
-  public static AbortingConditions of(Hub hub) {
-    AbortingConditions r = new AbortingConditions();
-    r.prepare(hub);
-    return r;
+    this.maxNonceReached = maxNonceReached;
   }
 
   public void reset() {
-    this.callStackOverflow = false;
-    this.insufficientBalance = false;
+    callStackOverflow = false;
+    insufficientBalance = false;
+    maxNonceReached = false;
   }
 
   public void prepare(Hub hub) {
-    this.callStackOverflow = hub.callStack().wouldOverflow();
+    callStackOverflow = hub.callStack().wouldOverflow();
     if (this.callStackOverflow) {
       return;
     }
 
-    this.insufficientBalance =
-        switch (hub.currentFrame().opCode()) {
+    final OpCode currentOp = hub.opCode();
+
+    insufficientBalance =
+        switch (currentOp) {
           case CALL, CALLCODE -> {
-            if (Exceptions.none(hub.pch().exceptions())) {
```

Comment:
I remove this unnecessary check, as we "prepare" the abortingConditions iff no Exceptions, so this checks was a duplicate

Reply:
No, what I mean is that we were having two times the check  that we don't have exceptions :  one before calling the `prepare` and once inside. So now we do it only once.

Reply:
I see